XenAdmin/Diagnostics/Checks/AssertCanEvacuateCheck.cs

```diff
@@ -78,8 +80,18 @@ protected List<Problem> CheckHost()
                     problems.Add(new LocalCD(this, residentVM));
                     VMsWithProblems.Add(residentVM.opaque_ref);
                 }
+
+                if (restrictMigration && residentVM.is_a_real_vm && !VMsWithProblems.Contains(residentVM.opaque_ref))

```

Comment:
For performance reasons, I believe this should be the first check-and-if-true-continue block. If the licence restricts migration, do we care about the other problems?